src/liger_kernel/transformers/functional.py

```diff
@@ -47,10 +48,20 @@ def liger_cross_entropy(
         reduction,
         softcap,
         return_z_loss,
+        return_token_accuracy,
     )
-    if not return_z_loss:
+
+    # Return based on what was requested
+    if not return_z_loss and not return_token_accuracy:
         return loss
-    return loss, z_loss
+
+    result = [loss]
+    if return_z_loss:
+        result.append(z_loss)
+    if return_token_accuracy:
+        result.append(token_accuracy)
+
+    return tuple(result) if len(result) > 1 else result[0]
```

Comment:
I see you tried to align with the code already present before your PR, but I find a potential ambiguity here: when a 2-tuple is returned, its 2nd element might be:
- `z_loss`: `return_z_loss=True, return_token_accuracy=False`
- `token_accuracy`: `return_z_loss=False, return_token_accuracy=True`